main.go

```diff
@@ -527,11 +526,11 @@ The validate command expects a configuration file as its only argument. Local fi
 		},
 		Action: func(c *cli.Context) error {
 			stage := c.Args().First()
-			config, err := config.Scan(collector.Directories(configScanDir...), collector.NoLogs)
+			config, err := agentConfig.Scan(collector.Directories(configScanDir...), collector.NoLogs)
 			if err != nil {
 				return err
 			}
-			cfg, err := elementalConfig.ReadConfigRunFromAgentConfig(config)
+			cfg, err := agentConfig.ReadConfigRunFromAgentConfig(config)
```

Comment:
this is returning a `v1.Config`, why do we still need that distinction vs `config` before? ... and if we have to keep it maybe it helps to name them something different to make this obvious

Reply:
umm, impossible, v1.Config does not exist anymore, there is now just one Config struct in the whole repo. Where did you see that?

Reply:
if I understand correctly 

```
config, err := agentConfig.Scan(collector.Directories(configScanDir...), collector.NoLogs)
```

see https://github.com/kairos-io/kairos-agent/pull/102/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R529 

returns a `config.Config`

while `cfg, err := agentConfig.ReadConfigRunFromAgentConfig(config)`

returns a types `v1.Config` which I saw was already replaced for some of the other cases

Reply:
ohh I see. Both are returning a config. There is no longer a v1.Config, but this is being called twice for no reason probably!

Reply:
and this returns the same config in both cases:

`func ReadConfigRunFromAgentConfig(c *Config) (*Config, error) `

Reply:
in any case, sometimes the config needs to be initialized in a different config to fullfill some of the required stuff.

for example the `agentConfig.Scan` will return a config, yes, but that config wont have the logger on it for example.

I tihnk the fix here is to hijack the Scan to create the config with the `Newconfig` method, so we dont have an empty config.Logger for example.

Reply:
good catch here, seems like I was doing this twice due to initializing the Config object on the scan directly. Now it does so all the config defaults are set from scratch and much more code can be removed, yay!

Reply:
ok, fixed and even improved!

Reply:
cool cool cool